pkg/ipcache/ipcache.go

```diff
@@ -69,12 +70,37 @@ type IPKeyPair struct {
 	Key uint8
 }
 
+const (
+	rLockWeight = 1
```

Comment:
This means we can have at most 2 readers now? Is it likely we will never have more than 2 readers?

Reply:
Given that we will have around 100 endpoints at a given time in the node would be a fare assumption that the maximum number of readers will be 100? cc @jrajahalme